### PR TITLE
Simplify `Tap#custom_remote?`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Extracted from https://github.com/Homebrew/brew/pull/16730 for easier review.

### CODE REVIEW

**Comment by apainintheneck** [comment]
Not sure this is really simpler.

**Comment by MikeMcQuaid** [comment]
Agreed. Find this hard to follow than the previous one and doesn't reduce branching. Sorry, passing on this.

**Comment by reitermarkus** [comment]
It's needed for Sorbet to work. Note the signature is added in a later commit in the original PR.

**Comment by reitermarkus** [comment]
Feel free to suggest an alternative that only calls `remote` once.

**Comment by reitermarkus** [comment]
Also note that `nonzero?` returns `nil` instead of `false`.

File: Library/Homebrew/tap.rb [hunk 1]
```diff
@@ -516,10 +516,9 @@ def uninstall(manual: false)
   end
 
   # True if the {#remote} of {Tap} is customized.
+  sig { returns(T::Boolean) }
   def custom_remote?
-    return true unless remote
-
-    remote.casecmp(default_remote).nonzero?
+    !remote&.casecmp(default_remote)&.zero?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/tap.rb:521]:
```suggestion
    return true unless remote

    !remote.casecmp(default_remote).zero?
```

**Comment by Bo98** [comment]
> Feel free to suggest an alternative that only calls `remote` once.

`return unless (remote = self.remote)` or similar?

File: Library/Homebrew/tap.rb (hunk 1 above)
Comment (MikeMcQuaid) [inline Library/Homebrew/tap.rb:521]:
```suggestion
    return true unless (tap_remote = remote)

    !tap_remote.casecmp(default_remote).zero?
```

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @reitermarkus!
